core/backend/src/test/annotations/TextAnnotation.test.ts

```diff
@@ -1241,5 +1283,77 @@ describe("TextAnnotation element", () => {
+      const logRange = (name: string, range: Range2d | Range3d) => {
+        console.log(name, ": \n\tJSON:", JSON.stringify(range.toJSON(), undefined, ""), "\n\txLength:", range.xLength(), "\n\tyLength:", range.yLength(), "\n\n");
+      }
+
+      const makeTextBlock = (props: TextBlockProps) => {
+        const textblock = TextBlock.create({ styleOverrides: { lineSpacingFactor: 0 }, ...props });
+        textblock.appendRun(makeTextRun("abc"));
+        textblock.appendRun(makeTextRun("defg"));
+        return textblock;
+      }
+
+      const getTransform = (origin: XYZ, range: Range2d, textAnnotation: TextAnnotation): Transform => {
+        // Compute the transform that positions and orients this annotation relative to its anchor point. Includes its rotation
+        const relativeTransform = textAnnotation.computeTransform(range);
+
+        // Now scoot that relative transform to origin
+        const origin3d = origin
+        const transform = Transform.createTranslation(origin3d);
+        return transform.multiplyTransformTransform(relativeTransform);
+      }
+
+      const test = (props: TextBlockProps) => {
+        // Create textblock
+        const block = makeTextBlock(props);
+
+        // Create annotation
+        const annotation = TextAnnotation.fromJSON({ textBlock: block.toJSON() });
+        annotation.anchor = { vertical: "middle", horizontal: "center" };
+        const el0 = createElement();
+        el0.setAnnotation(annotation);
+
+        // Insert element
+        const elId = el0.insert();
+
+        // Get element
+        const el1 = imodel.elements.getElement<TextAnnotation3d>(elId);
+        const anno = el1.getAnnotation();
+        const newTextBlock = anno?.textBlock;
+        expect(newTextBlock).not.to.be.undefined;
+
+        // do layout on annotation1 and annotation2
+        const newLayout = doLayout(newTextBlock!);
+        const transform = getTransform(el1.placement.origin, newLayout.range, anno!);
+
+        // Dubugging the ranges
+        logRange("el0 bbox", el0.placement.bbox);
+        logRange("el1 bbox", el1.placement.bbox);
+        logRange("newLayout.range", newLayout.range);
+        logRange("transformed newLayout.range", transform.multiplyRange(Range3d.fromJSON(newLayout.range)));
+        // logRange("newLayout.marginRange", newLayout.marginRange);
+        // logRange("transformed newLayout.marginRange", transform.multiplyRange(Range3d.fromJSON(newLayout.marginRange)));
+
+        // Check that the bounding box is correct
+
+        // Before my changes, this test should pass. After my changes, it should fail.
+        const range3d = Range3d.fromJSON(newLayout.range);
+        expect(el1.placement.bbox.isAlmostEqual(range3d)).to.be.true;
+
+        // After my changes, this should pass instead.
+        // const range3d = Range3d.fromJSON(newLayout.marginRange);
+        // expect(el1.placement.bbox.isAlmostEqual(range3d)).to.be.true;
+      };
+
+      // 0 margins
```

Comment:
Broken test I mentioned.

Reply:
> it doesn't match the original range

By "original range" are you referring to this null range?
> `JSON: {"low":[1e+200,1e+200,1e+200],"high":[-1e+200,-1e+200,-1e+200]}`

Reply:
I mean that the `el1` range: `JSON: {"low":[-2.766357421875,-1,0],"high":[2.766357421875,1,0]}` does not match the `TextBlockLayout` range: `JSON: [[0,-1],[7,0]]`

I suppose the el0 range `isNull` because it hasn't been inserted into the model yet?

Reply:
(I think I was calling the original range the `TextBlockLayout` range computed from `doLayout`)

Reply:
> I suppose the el0 range isNull because it hasn't been inserted into the model yet?

Right.

> the el1 range: JSON: {"low":[-2.766357421875,-1,0],"high":[2.766357421875,1,0]} does not match the TextBlockLayout range: JSON: [[0,-1],[7,0]]

Thanks for clarifying. The element's bounding box is computed from the bounding boxes of the `TextString`(s) inserted into its geometry stream, which are created from the `TextBlock`. Some [padding is added](https://github.com/iTwin/imodel-native/blob/5bae73c8872eea691dc6e1b36d0935293c366305/iModelCore/iModelPlatform/DgnCore/ElementGeometry.cpp#L313).

Reply:
Okay, then this is as expected. I wasn't sure. So should bounding box be specifically for the geometry actually produced, and if the user wants something more explicit, use the `TextBlockLayout.range` or `TextBlockLayout.marginRange` depending on use-case? 

If so, I can leave a comment somewhere about that and delete this test.

Reply:
> should bounding box be specifically for the geometry actually produced

Yes. If at some point you start adding boxes or other kinds of frames, those should produce additional geometry that would expand the element's bounding box, accounting for the margins. When you're only producing `TextString` geometry, the size of the bounding box should be indifferent to the margins. The offsets of the `TextString`s as computed during layout would be affected by the margins.

Reply:
Okay, sounds good. Whenever I do that, I can add these tests back for that specific use case. As it is, I'll remove this test for now.